sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -142,7 +142,7 @@ case class CreateTableAsSelect[T](
     tableName: String,
     child: LogicalPlan,
     allowExisting: Boolean,
-    desc: Option[T] = None) extends UnaryNode {
+    desc: T) extends UnaryNode {
```

Comment:
I add a specific `CreateHiveTableAsSelect` but still keep`CreateTableAsSelect` as a common logical plan for the use of other SQL dialects.